src/lib/y2network/route.rb

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+module Y2Network
+  # This class represents a network route
+  class Route
+    # @return [IPAddr]
+    attr_reader :to
+    # @return [Interface]
+    attr_reader :interface
+    # @return [IPAddr,nil]
+    attr_reader :source
+    # @return [IPAddr,nil]
+    attr_reader :gateway
+
+    def initialize(to, interface, gateway: nil, source: nil, preference: nil)
+      @to = to
+      @interface = interface
+      @gateway = gateway
+      @source = source
+      @preference = preference
```

Comment:
preferences is current options?

Reply:
No, it is the priority (as it is called in the `ip route` man page). To be honest, we do not need it right  now, I was just sketching how the API could look like if it was needed.

Reply:
Probably one of the options, so suggested a rename

Reply:
so I worry we will really need generic options here. I know it is expert only, but especially in autoyast many customer can depend on it to configure specific routing parameter which we are not even aware they use.